autoload/ale/definition.vim

```diff
@@ -17,6 +20,18 @@ function! ale#definition#ClearLSPData() abort
     let s:go_to_definition_map = {}
 endfunction
 
+function! ale#definition#UpdateTagStack() abort
+    let l:should_update_tagstack = exists('*gettagstack') && exists('*settagstack') && g:ale_update_tagstack
+
+    if l:should_update_tagstack
+        let l:from = [bufnr('%'), line('.'), col('.'), 0]
```

Comment:
You can use the filename, line, and column we already have to determine this.

Reply:
Assuming you're referring to `l:filename`, `l:line` and `l:column` defined in `ale#definition#HandleTSServerResponse` and `ale#definition#HandleLSPResponse` - these are the targets for the new location.  The values I'm pushing on to the stack are based on the old location.

I'll change the variable names to be a bit more clear.  :)